src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/CryptoHelper.cs

```diff
@@ -0,0 +1,442 @@
+                    // to a custom algorithm with keySize 192 bits. If he does that, we anyways make sure that we return false.
+            }
+        }
+
+        internal static void FillRandomBytes(byte[] buffer)
+        {
+            RandomNumberGenerator.GetBytes(buffer);
+        }
+
+        /// <summary>
+        /// This generates the entropy using random number. This is usually used on the sending 
+        /// side to generate the requestor's entropy.
+        /// </summary>
+        /// <param name="data">The array to fill with cryptographically strong random nonzero bytes.</param>
+        public static void GenerateRandomBytes(byte[] data)
+        {
+            RandomNumberGenerator.GetNonZeroBytes(data);
+        }
+
+        /// <summary>
+        /// This method generates a random byte array used as entropy with the given size. 
+        /// </summary>
+        /// <param name="sizeInBits"></param>
+        /// <returns></returns>
+        public static byte[] GenerateRandomBytes(int sizeInBits)
+        {
+            int sizeInBytes = sizeInBits / 8;
+            if (sizeInBits <= 0)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException(nameof(sizeInBits), SR.Format("ID6033", sizeInBits)));
+            }
+            else if (sizeInBytes * 8 != sizeInBits)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentException(SR.Format("ID6002", sizeInBits), nameof(sizeInBits)));
+            }
+
+            byte[] data = new byte[sizeInBytes];
+            GenerateRandomBytes(data);
+
+            return data;
+        }
+
+        internal static RandomNumberGenerator RandomNumberGenerator
+        {
+            get
+            {
+                if (s_random == null)
+                {
+                    s_random = RandomNumberGenerator.Create();
+                }
+                return s_random;
+            }
+        }
+
+        internal static HashAlgorithm CreateHashAlgorithm(string algorithm)
+        {
+            object algorithmObject = GetAlgorithmFromConfig(algorithm);
+
+            if (algorithmObject != null)
+            {
+                HashAlgorithm hashAlgorithm = algorithmObject as HashAlgorithm;
+                if (hashAlgorithm != null)
+                {
+                    return hashAlgorithm;
+                }
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new InvalidOperationException(SR.Format(SR.CustomCryptoAlgorithmIsNotValidHashAlgorithm, algorithm)));
+            }
+
+            switch (algorithm)
+            {
+                case SHAString:
+                case SHA1String:
+                case SystemSecurityCryptographySha1String:
+                case SecurityAlgorithms.Sha1Digest:
+#pragma warning disable CA5350 // Do not use insecure cryptographic algorithm SHA1.
```

Comment:
Remove all #pragma's for code's which start with CA. These were build static analysis tools used for .NET Framework. Both the disable and the restore pragma's need to be removed